lib/travis/sso/generic.rb

```diff
@@ -99,8 +102,14 @@ def allow(request)
 
         def login(request)
           return unless token = sso_token(request)
-          raw  = open("#{endpoint}/users?access_token=#{token}", 'Accept' => accept)
-          data = MultiJson.decode(raw.read)
+          uri = URI.parse("#{endpoint}/users?access_token=#{token}")
+          http = Net::HTTP.new(uri.host, 443)
+          http.use_ssl = true
+          http.verify_mode = OpenSSL::SSL::VERIFY_NONE if enterprise
```

Comment:
not 'if enterprise' -> enterprise installation doesn't mean it will always use self-signed certs